### ISSUE TITLE
Convert Task to a tree (NestedSet)

### ISSUE THREAD

**Issue by rmehta** [frappe/erpnext#11014]
- [x] Add tree view for task
- [ ] Project can also become a task (?)

#### Task-tree enhancements
- [ ] renaming should change the subject of task
- [ ] Task can only depend on tasks of the same parent group

**Comment by giuseppelamatrice** [comment]
In "building & construction" is very used use Project management. 
What do you think of this?
- Duplicate Project
- Create a Sub-Project with all task.
- Link Sub-Project to Master Project.

**Comment by Zlash65** [comment]
Should there be a feature like - A Task group is given a start and end date and each task within this group should be panned out such that all these tasks are completed before the groups range ( ? )

**Comment by rmehta** [comment]
@Zlash65 first convert into tree. next steps later!

**Comment by rmehta** [comment]
@giuseppelamatrice the tree will solve the issue for multi layout. What you are proposing is to hard to code it to one level. And yes we want to set this up for building construction projects!

**Comment by jigneshpshah** [comment]
@nabinhait , So is this not going to be developed in November milestone ?

**Comment by rmehta** [comment]
This is already done and merged in develop

**Comment by jigneshpshah** [comment]
Excellent ! :-)

**Comment by jigneshpshah** [comment]
I tried with the develop branch but unable to save new project.

Error trace as below 

> 
>   File "/home/ashish/frappe-bench/apps/erpnext/erpnext/projects/doctype/task/task.py", line 64, in update_nsm_model
>     frappe.utils.nestedset.update_nsm(self)
>   File "/home/ashish/frappe-bench/apps/frappe/frappe/utils/nestedset.py", line 39, in update_nsm
>     if not doc.lft and not doc.rgt:
> AttributeError: 'Task' object has no attribute 'lft'

**Comment by ashish-greycube** [comment]
@Zlash65 , above issue is solve by running following command
bench update --patch

**Comment by Zlash65** [comment]
Let me know if there are any issues still remaining to tackle.

**Comment by ashish-greycube** [comment]
@Zlash65 , is this fix dependent on any other fix or PR. I mean I have a installation from master branch (not develop branch) and if i apply the 7 file changes done by you , am i good to go? thanks
Is there a way to selectively consume this changes only and not entire develop branch changes

**Comment by Zlash65** [comment]
@ashish-greycube Yes it is dependent on other PRs. A lot of refactoring was put into motion after this merge in both frappe as well as erpnext side.

**Comment by jigneshpshah** [comment]
@Zlash65  any idea when this feature will get merged in to master branch ? Eagerly waiting for the same !
